### PR TITLE
fixed Registration tool email field to accept + in email

### PR DESCRIPTION
**Pull request by rohnsha0**

### CODE REVIEW

**Comment by rohnsha0** [comment]
@mauritsvanrees @jensens pleasse check if it breaks anything?

**Comment by rohnsha0** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/tests/testEmailLogin.py [hunk 1]
```diff
@@ -83,8 +83,8 @@ def testEmailMemberIdsAllowed(self):
         # email addresses.
         # A plus sign in the id gives problems in some parts of the
         # UI, so we do not allow it.
-        self.assertFalse(pattern.match("user+test@example.org"))
-        self.assertFalse(registration.isMemberIdAllowed("user+test@example.org"))
+        #self.assertFalse(pattern.match("user+test@example.org")) 
+        #self.assertFalse(registration.isMemberIdAllowed("user+test@example.org"))
```
Comment (mauritsvanrees) [inline Products/CMFPlone/tests/testEmailLogin.py:87]: Can you change this so it tests that these email address *are* allowed?  So change `assertFalse` in `assertTrue`.  And then the comment can be removed.

File: news/3968.bugfix [hunk 2]
```diff
@@ -0,0 +1,2 @@
+fixed RgistrationTool to take useremail with "__+__@abc.com"
```
Comment (mauritsvanrees) [inline news/3968.bugfix:1]:
```suggestion
Fixed RegistrationTool to take user email with `__+__@abc.com`.
```

**Review by mauritsvanrees — COMMENTED** [review]
Will try later.  For now two small inline remarks to address.

Also, since you are a member of the GitHub plone organisation, you can also create a branch here, instead of doing it in a fork.  Then others can help by committing directly on your branch, if needed.  But never mind for now, no problem, just a suggestion.

File: Products/CMFPlone/tests/testEmailLogin.py (hunk 1 above)
Comment (rohnsha0) [reply Products/CMFPlone/tests/testEmailLogin.py]: please verify @mauritsvanrees

**Comment by rohnsha0** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/tests/testEmailLogin.py [hunk 3]
```diff
@@ -83,8 +83,9 @@ def testEmailMemberIdsAllowed(self):
         # email addresses.
         # A plus sign in the id gives problems in some parts of the
         # UI, so we do not allow it.
```
Comment (mauritsvanrees) [inline Products/CMFPlone/tests/testEmailLogin.py:85]: Can you remove this comment?  It is outdated after your changes.

Comment (rohnsha0) [reply Products/CMFPlone/tests/testEmailLogin.py]: done

**Comment by mauritsvanrees** [comment]
Note that you can also go to the `portal_registration` tool in the ZMI and then the Configure panel:
http://localhost:8080/Plone/portal_registration/manage_editIDPattern
There you can specify an alternative user id pattern:

Screenshot 2024-06-14 at 17 38 27

I tried it and it still works.  So if you want this in your own site, you can simply change the pattern in the ZMI.

**Comment by rohnsha0** [comment]
>So we would need a similar `makeQuery` call on the author page.

using urlib? can we change the query func? @mauritsvanrees

**Comment by rohnsha0** [comment]
in below code this_user is groupMember but what to get in case of username previously in author.pt we have view/username... now?
https://github.com/plone/Products.CMFPlone/blob/f7a537773cae9c4b56f57e31ac02192a7ce35648/Products/CMFPlone/controlpanel/browser/usergroups_groupmembership.pt#L133

**Comment by mauritsvanrees** [comment]
In `author.pt` it was:

    Creator=${username}

So now it should be:

    Creator=${view.makeQuery(username)}

`makeQuery` is defined here:

https://github.com/plone/Products.CMFPlone/blob/f7a537773cae9c4b56f57e31ac02192a7ce35648/Products/CMFPlone/controlpanel/browser/usergroups.py#L71-L72

So you would need to copy that method (and the import it needs) to the [AuthorView class](https://github.com/plone/Products.CMFPlone/blob/f7a537773cae9c4b56f57e31ac02192a7ce35648/Products/CMFPlone/browser/author.py#L128).

I think that should be it.

**Comment by rohnsha0** [comment]
@mauritsvanrees kindly verify the changes

File: Products/CMFPlone/browser/templates/author.pt [hunk 4]
```diff
@@ -219,7 +219,7 @@
 
                           <p>
                           <a href=""
-                             tal:attributes="href string:$here_url/search?Creator=${username}&amp;sort_on=created&amp;sort_order=reverse"
+                             tal:attributes="href string:$here_url/search?Creator=${view.makeQuery(username)}&amp;sort_on=created&amp;sort_order=reverse"
                              i18n:translate="go_to_search_author_content">
```
Comment (mauritsvanrees) [inline Products/CMFPlone/browser/templates/author.pt:223]: This gives an error when I view the author page.  Also, we were using an older way of spelling this, using `tal:attributes`, which can be made shorted and more readable.  Can you change it to the following?  Then it works for me.

```
                          <a href="${here_url}/search?${python:view.makeQuery(Creator=username, sort_on='created', sort_order='reverse')}"
                             i18n:translate="go_to_search_author_content">
```

Comment (rohnsha0) [reply Products/CMFPlone/browser/templates/author.pt]: @mauritsvanrees now please verify

**Comment by mauritsvanrees** [comment]
Yes, now it is fine.  Thanks!

**Review by mauritsvanrees — APPROVED** [review]

**Pull request merged**